src/pack_unpack.c

```diff
@@ -29,6 +29,7 @@ typedef struct {
     int line;
     int column;
     size_t pos;
+    int scan_mode;
```

Comment:
I don't like the `scan_mode` name. It should reflect the fact that an error was encountered and "error recovery" is in progress.

Reply:
`int error_recovery;` ?

Reply:
Yeah, or maybe just `has_error`